### DISCUSSION TITLE
How to do a Tippy.js integration

### DISCUSSION THREAD

**Discussion by Ryuuji159** [livewire/livewire#7711] [1. Help] [answered]
I'm trying to integrate Tippy.js the following way, that is executed on DOMContentLoaded
```js
tippy('[tooltip]', {
      arrow: false,
      theme: 'translucent',
      content: (reference) => reference.getAttribute('tooltip'),
  });
```
That way every element that has a tooltip will show it with a tippy tooltip.

But I dont know how to handle page changes, since I'm using full page components and navigation, there is never really a new page load, so when I navigate to other page, tippy doesnt have a chance to reload.

Right now I'm calling the following to solve it, but it feels like a waste, because it creates a new element everytime, and needs to iterate over every element that has been created, and I dont have a place to do a cleanup of old tippy objects.
 ```js
Livewire.hook('element.init', ({component, el}) => {
      if (el.hasAttribute('tooltip')) {
          tippy(el, {
              arrow: false,
              theme: 'translucent',
              content: (reference) => reference.getAttribute('tooltip'),
          });
      }
  })
```

I think i need an event that gets executed after everything is loaded on the page, but can seem to find it. I really would appreciate some help here.

**Comment by danie-ramdhani** [comment, marked as answer]
try this package:

https://github.com/ryangjchandler/alpine-tooltip

**Reply by Ryuuji159** [reply]
That works really well, Im using it on my project now

**Comment by d-shannon** [comment]
```
 document.addEventListener('livewire:navigated', () => {
// hook here
});
```
